pkg/controller/plan/kubevirt.go

```diff
@@ -1507,7 +1507,7 @@ func (r *KubeVirt) ensureSecret(vmRef ref.Ref) (secret *core.Secret, err error)
 }
 
 // Build the DataVolume credential secret.
-func (r *KubeVirt) secret(vmRef ref.Ref) (object *core.Secret, err error) {
+func (r *KubeVirt) secret(vmRef ref.Ref, populator bool) (object *core.Secret, err error) {
```

Comment:
wouldn't it be nicer to pass a function that sets the data instead of propagating the arguments all the way down here?

Reply:
i think it will just complex it. it's ~1 liner at the moment. instead of propagating this boolean we will propagate the function...

Reply:
I think differently - it would be better to generalize this method with dependency injection rather than adding more code flows inside it, I'll post a separate PR to talk about the concrete changes I have in mind